app/utils/checkOrigin.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+const {parse} = require('url');
+
+const {default: logger} = require('@financial-times/n-logger');
+
+const defaultMethods = [
+    'GET',
+    'POST',
+    'PUT',
+    'DELETE'
+];
+
+const defaultDomain = 'ft.com';
+
+module.exports = ({
+    methods = defaultMethods,
+    domain = defaultDomain
+} = {}) => (request, response, next) => {
+
+    const checkDomain = url =>
+        Boolean(url && parse(url).hostname.endsWith(`.${domain}`));
+
+    if (response.locals && response.locals.flags && response.locals.flags.disableSudsOriginCheck) {
+        return next();
+    }
+
+    const {
+        originalUrl,
+        method
+    } = request;
+
+    if (!methods.includes(method)) {
+        return next();
+    }
+
+    const origin = request.get('origin');
+    const referer = request.get('referer');
+
+    const originIsGood = checkDomain(origin);
```

Comment:
This should be done only if origin exists, so it could be added in the statement below, something like...
```
if (!origin) {
  log...
} else {
   ...check the origin
   if (!originIsGood) {
       ...log...
   } else {
       return next();
  }
}
```

Reply:
I deliberately try to avoid lots of nested `if`s, and `originIsGood` guards against undefined origin. But I accept this is a coding style thing, so not too bothered about it.